src/AuthenticatedApp.tsx

```diff
@@ -59,13 +59,18 @@ const Wrapper: FunctionComponent<RouteComponentProps & {sessionData: UserSession
 
 const AuthenticatedApp: FunctionComponent<RouteComponentProps> = ({location, match}) => {
   const sessionData = useUserSessionDataState()
+  const sessionUpdateFn = useUserSessionDataDispatch()
 
   if (!sessionData.token) {
-    //save location and redirect
-    if (location.pathname !== '/') {
-      sessionStorage.setItem('location', `${location.pathname}${location.search}`)
+    if (sessionData.lastLoginMethod === 'OAUTH_SYNAPSE') {
+      //save location and redirect
+      if (location.pathname !== '/') {
+        sessionStorage.setItem('location', `${location.pathname}${location.search}`)
+      }
+      Utility.redirectToSynapseLogin()
+    } else {
+      sessionUpdateFn({type: 'LOGOUT'})
     }
```

Comment:
Continue redirecting to OneSage for users that signed via Synapse OAuth, but log out all other users.

Reply:
Can you add a TODO comment (signed and dated) with the JIRA ticket for allowing reauth: https://sagebionetworks.jira.com/browse/DHP-1057